File: csharp/src/RespBench/Command/Impl/GetCommand.cs
/*
 * Copyright 2025 the original author or authors.
 */
using RespBench.Client;
using RespBench.Config;
using RespBench.Engine;

namespace RespBench.Command.Impl;

public class GetCommand : ICommand
{
    private double _weight = 1.0;

    public string Name => "GET";
    public string Description => "GET key";
    public double Weight => _weight;

    public void Configure(CommandConfig config)
    {
        _weight = config.Weight;
    }

    public async Task<CommandResult> Execute(IBenchmarkClient client, KeyGenerator keyGenerator)
    {
        byte[] key = keyGenerator.NextKey();
        try
        {
            var result = await client.Get(key).ConfigureAwait(false);
            return CommandResult.SuccessResult("GET", result.LatencyMicros);
        }
        catch (Exception e)
        {
            return CommandResult.Failure("GET", 0, e.Message);
        }
    }
}

File: php/src/Command/Impl/GetCommand.php
<?php

declare(strict_types=1);

namespace RespBench\Command\Impl;

use RespBench\Client\BenchmarkClient;
use RespBench\Command\Command;
use RespBench\Command\CommandResult;
use RespBench\Engine\KeyGenerator;

final class GetCommand extends Command
{
    public function execute(BenchmarkClient $client, KeyGenerator $keyGenerator): CommandResult
    {
        $key = $keyGenerator->nextKey();
        $result = $client->get($key);

        return new CommandResult($this->name, $result->latencyMicros, $result->isSuccess());
    }
}
